corehq/apps/data_cleaning/models.py

```diff
@@ -1010,6 +1010,22 @@ def select_multiple(self, session, doc_ids):
             is_selected=False,
         ).update(is_selected=True)
 
+    @retry_on_integrity_error(max_retries=3, delay=0.1)
+    @transaction.atomic
+    def deselect_multiple(self, session, doc_ids):
+        # update is_selected to False for all selected records that have changes
+        session.records.filter(
+            doc_id__in=doc_ids,
+            is_selected=True,
+            changes__isnull=False,
+        ).update(is_selected=False)
+
+        # delete all records that have no changes
+        session.records.filter(
+            doc_id__in=doc_ids,
+            changes__isnull=True,
+        ).delete()
```

Comment:
It may be that `session` is unnecessary here. I believe `self` already contains however the object manger is being called, so `self.filter(...)` should function the same as `session.records.filter(...)` is this is being called like `session.records.deselect_multiple(...)`. But we should verify that this is the case before making any changes